### PR TITLE
feat(arc-515): add permissions table

### PR DESCRIPTION
**Pull request by BavoVdg**
[ARC-515](https://meemoo.atlassian.net/browse/ARC-515)

UPDATE: https://github.com/viaacode/hetarchief-client/pull/381

- Permissions table

image

### CODE REVIEW

**Review by bertyhell — APPROVED** [review]

**Comment by bertyhell** [comment]
looks good
only the save and cancel buttons will probably need to be rendered in the client in the header bar. Similar to how it is done in the content page edit page:
image

File: demo/src/react-admin/modules/user-group/const/user-group.const.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		dataType: 'dateTime',
+	},
+	{
+		id: 'actions',
+		tooltip: Config.getConfig().services.i18n.t('admin/user-groups/user-group___acties'),
+		visibleByDefault: true,
+	},
+];
+
+export enum SpecialUserGroup {
```
Comment (bertyhell) [inline demo/src/react-admin/modules/user-group/const/user-group.const.tsx:137]: this is duplicated in 2 files:
image

File: src/modules/user-group/services/user-group.service.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	// 	} catch (err) {
+	// 		throw new CustomError('Failed to remove permission groups from user group', err, {
+	// 			query: 'REMOVE_PERMISSION_GROUPS_FROM_USER_GROUP',
+	// 			variables: {
+	// 				permissionGroupIds,
+	// 				userGroupId,
+	// 			},
+	// 		});
+	// 	}
+	// }
```
Comment (bertyhell) [inline src/modules/user-group/services/user-group.service.ts:147]: are these still needed, or can they be removed?

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: demo/src/react-admin/modules/user-group/const/user-group.const.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+export const UserGroupTableColumns = (
+	userGroups: UserGroupArchief[],
+	updateUserGroup: (groupId: string, permissionId: string, value: boolean) => void,
+): (Column<PermissionData> & UseSortByColumnOptions<PermissionData>)[] => [
+	{
+		Header: '',
+		accessor: 'label',
+		disableSortBy: true,
+	},
+	...GET_USER_GROUPS().map((groupLabel) => {
```
[the remark this answers is not in the document]
Comment (BavoVdg) [reply demo/src/react-admin/modules/user-group/const/user-group.const.tsx:61]: Om translations aan de labels toe te voegen, maar ik kan in principe het label uit de database gebruiken!

File: src/modules/user-group/services/user-group.service.ts (hunk 2 above)
Comment (BavoVdg) [reply src/modules/user-group/services/user-group.service.ts:147]: Removed.

File: demo/src/react-admin/modules/user-group/const/user-group.const.tsx (hunk 1 above)
Comment (BavoVdg) [reply demo/src/react-admin/modules/user-group/const/user-group.const.tsx:137]: Hier waren idd een aantal duplicates. Fixed.

File: demo/src/react-admin/modules/user-group/const/user-group.const.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (BavoVdg) [reply demo/src/react-admin/modules/user-group/const/user-group.const.tsx]: Fixed.

**Review by bertyhell — APPROVED** [review]

**Pull request merged**